pkg/controller/perconaservermongodb/status.go

```diff
@@ -101,8 +107,7 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 	}
 
 	cr.Status.State = api.AppStateInit
-	if replsetsReady == len(cr.Spec.Replsets) && clusterState == clusterReady {
-
+	if replsetsReady == len(repls) && clusterState == clusterReady {
```

Comment:
just a comment, at the current moment cluster state does not include mongos pod state

Reply:
Yeah, we do not have it. But it will be a separate task.